wallet/src/adapters/http.rs

```diff
@@ -45,12 +45,12 @@ impl WalletCommAdapter for HTTPWalletCommAdapter {
 			error!("{}", err_str,);
 			Err(ErrorKind::Uri)?
 		}
-		let url = format!("{}/v1/wallet/foreign/receive_tx", dest);
+		let url = dest.to_owned() + "/v1/wallet/foreign/receive_tx";
```

Comment:
for my education - why do you prefer `+`?

Reply:
Actually in this case there shouldn't be any meaningful difference except for readability.
But in general, the `format!` macro is almost always less efficient than pushing, it dismisses any pre allocated memory and just copies everything to a new memory location.

But here because `dest` is `&str` and not `String` it's not really less efficient and I probably did it out of habit(I use format only on non string variables), but I shouldn't have changed someone elses code for no good enough reason. 

btw, even with `&str`s `format!` is just a little slower than anything else https://github.com/hoodie/concatenation_benchmarks-rs

Reply:
Seems to be some kind of premature optimization to me at this stage.

Reply:
@mably you're right. If it's already a String I think you shouldn't use `format!`, you should use `push` or `+`,  because that's less idiomatic and multiple redundant allocations.
But if it's a `&str` anyway that's really a premature optimization and I won't change that in the future.

Reply:
If no performance improvement, or simply the code, I prefer not changing those trivial things.

And agree with @mably , better to focus on functional issues at this stage.

Reply:
I can try and amend it out if you prefer